### DISCUSSION TITLE
Appropriate for large datasets (many items)

### DISCUSSION THREAD

**Discussion by mgrivera** [table-library/react-table-library#14] [Q&A] [answered]
Hi Robin, 
Thanks for all your work in blogs, books, courses and, now, in this particular library. 

Just wondering if the table (lib) is designed to manage large quantity of items. At least for me, a large list is not so large. Some 1000 - 2000 items is a range I considered a large list for a table. 

So the question is: is the library designed with this feature in mind? 

Again, many thanks for your work.

**Comment by rwieruch** [comment]
Hi @mgrivera and thanks for your interest. 

To be honest, I didn't try the table with lots of data **which is displayed at once** yet. Even though we are using the table with lots of data (up to 100.000 nodes), its paginated (see [here](https://www.robinwieruch.de/react-tree-list/)). 

Anyway, since the table is composed of components, I could imagine that you could maybe just wrap the Body component in something like React Virtualized or React Window? Would be worth a try.

Sorry that I don't have an example for this case yet. I know it's a common problem, but I extended the table library only so far to meet my client's requirements 😅

**Reply by manuelguillermo** [reply]
Thanks for your answer Robin, it is very appreciated ...

**Reply by rwieruch** [reply]
So we came to a point ourselves where we had to use React Window. It works and I try to write up an example over the week!

**Comment by rwieruch** [comment, marked as answer]
Story: https://react-table-library.com/?path=/story/recipes-large-tables--large-list-tree
Source Code: https://github.com/table-library/react-table-library/blob/master/.storybook/stories/Recipes/large-table.story.js
